### DISCUSSION TITLE
How to set ACA authentication to the container app when deploying

### DISCUSSION THREAD

**Discussion by BartNetJS** [microsoft/aspire#2016] [Q&A] [answered]
I'm protecting my ACA container with authentication on the edge:
![image](https://github.com/dotnet/aspire/assets/4865355/ac8c10cc-c61e-438b-ad8b-0ada0d779142)
How can i enable this in my bicep or manifest file by default?
Currently after a deploy I go to the azure portal and enable authentication.

**Comment by davidfowl** [comment]
Currently when you want to configure anything that isn't the default, you have to drop to bicep. https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?tabs=windows#generate-bicep-from-net-aspire-app-model

We're trying to come up with a model to reduce the need for this cliff but this is what you'll have to do for the next couple of previews.

**Comment by BartNetJS** [comment]
I tried that path, but currently the manifest files are not generate (except for the files in the infra folder)
See my comment started here: https://github.com/Azure/azure-dev/issues/3179#issuecomment-1911338691

**Comment by davidfowl** [comment]
Thre's no other way to change the deployment when using aspire.

**Comment by davidfowl** [comment, marked as answer]
In 9.2 you should be able to configure this https://learn.microsoft.com/en-us/dotnet/aspire/whats-new/dotnet-aspire-9.2#-configure-azure-container-apps-environments
